### ISSUE TITLE
Prework review

### ISSUE THREAD

**Issue by avo1** [vulong264/iosprework#1]
Thanks for you submission. 👍 nice work with Tip Calculator.

### Quick Comments:

- Good usage of `UserDefaults`. This is one of the methods to pass the data from multiple viewController. We will learn few more ways to do that. The better place to save data is as soon as the user key in. Don't wait for `viewWillDisappear`, it may be too late.
- Xcode is complaining here. You can use `if let` instead.

<img width="752" alt="screen shot 2017-02-08 at 8 41 17 pm" src="https://cloud.githubusercontent.com/assets/10482065/22739672/30f21534-ee3f-11e6-9c98-b19a536b8998.png">

- Nice try with the table view. 
- Always a good practice to validate user input. Don't let user enter invalid data like this

<img width="281" alt="screen shot 2017-02-08 at 8 35 54 pm" src="https://cloud.githubusercontent.com/assets/10482065/22739764/b4b7cf8a-ee3f-11e6-8120-3e24b87180cd.png">

- In the setting page, in custom mode, you can validate the input to ensure bad < normal < good.
- When you embed your view controller into the Navigation Controller, you don't need to manually add the `Back` button. But if you still want to do it, you will use `popViewController`, instead of `dismiss`. So when to use what? We'll learn later.

<img width="435" alt="screen shot 2017-02-08 at 8 56 10 pm" src="https://cloud.githubusercontent.com/assets/10482065/22740073/f16be230-ee40-11e6-8d0e-3c8bd1d24b5b.png">

### Optional Suggestions to Improve

* It's good that you have the list of countries. You can try `NSNumberFormatter` and `CurrencyStyle` to make your app adapt with multiple regions.

### Next step

Let's schedule a quick call to confirm that this course is a right fit for you. 
We also would like to know a little more about your experience. 
Please book a slot from [here](https://calendly.com/harley/15m).

**Comment by vulong264** [comment]
Hi Dave,

Thanks for your review and feedback! There's so much to learn for me!
Just FYI, I base in Hanoi and I know that this course will be held in SG. As discussed with Harley, I applied so that we can see how CoderSchool should handle remote students effectively. 

-- 
Vu Long
Mobile:+84.989.099.338
Skype: mr.vulong

On February 8, 2017 at 8:57:21 PM, Vo Dave (notifications@github.com) wrote:

Thanks for you submission. 👍 nice work with Tip Calculator.

Quick Comments:

Good usage of UserDefaults. This is one of the methods to pass the data from multiple viewController. We will learn few more ways to do that. The better place to save data is as soon as the user key in. Don't wait for viewWillDisappear, it's may be too late.
Xcode is complaining here. You can use if let instead.


Nice try with the table view.
Always a good practice to validate user input. Don't let user enter invalid data like this


In the setting page, in custom mode, you can validate the input to ensure bad < normal < good.
When you embed your view controller into the Navigation Controller, you don't need to manually add the Back button. But if you still want to do it, you will use popViewController, instead of dismiss. So when to use what? We'll learn later.


Optional Suggestions to Improve

It's good that you have the list of countries. You can try NSNumberFormatter and CurrencyStyle to make your app adapt with multiple regions.
Next step

Let's schedule a quick call to confirm that this course is a right fit for you.
We also would like to know a little more about your experience.
Please book a slot from here.

**Comment by avo1** [comment]
Yes I still remember you and was surprised seeing your submission. I asked Harley and he explained to me.
